File: Professional Image Modifier 4/[VB]-Professional Image Modifier 4/VB/ImageFunctions/Controls/HarrisCornerProperties.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace ImageFunctions.Controls
	Partial Public Class HarrisCornerProperties
		Inherits UserControl

		Public Property SigmaDefault() As Integer
		Public Property ThresholdDefault() As Integer
		Public Property Sigma() As Integer
		Public Property Threshold() As Integer

		''' <summary>
		''' The Defaults are not real defaults - you should investigate appropriate defaults for Threshold and Sigma
		''' </summary>
		Public Sub New()
			InitializeComponent()
			ThresholdDefault = 5
			SigmaDefault = 3
			trackThreshold.Value = ThresholdDefault
			trackSigma.Value = SigmaDefault
		End Sub

		Public Sub SetDefaults()
			trackThreshold.Value = ThresholdDefault
			trackSigma.Value = SigmaDefault
		End Sub


		Private Sub trackThreshold_ValueChanged_1(ByVal sender As Object, ByVal e As EventArgs) Handles trackThreshold.ValueChanged
			lblThreshold.Text = trackThreshold.Value.ToString()
			Threshold = trackThreshold.Value
		End Sub

		Private Sub trackSigma_ValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trackSigma.ValueChanged
			lblWindow.Text = trackSigma.Value.ToString()
			Sigma = trackSigma.Value
		End Sub
	End Class
End Namespace

File: Professional Image Modifier 4/[C#]-Professional Image Modifier 4/C#/ImageFunctions/Controls/HarrisCornerProperties.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ImageFunctions.Controls
{
	public partial class HarrisCornerProperties : UserControl
	{

		public int SigmaDefault { get; set; } // As set by Aforge
		public int ThresholdDefault { get; set; }  // As set by Aforge
		public int Sigma { get; set; } // Set by User
		public int Threshold { get; set; } // Set by User

		/// <summary>
		/// The Defaults are not real defaults - you should investigate appropriate defaults for Threshold and Sigma
		/// </summary>
		public HarrisCornerProperties()
		{
			InitializeComponent();
			ThresholdDefault = 5;
			SigmaDefault = 3;
			trackThreshold.Value = ThresholdDefault;
			trackSigma.Value = SigmaDefault;
		}

		public void SetDefaults()
		{
			trackThreshold.Value = ThresholdDefault;
			trackSigma.Value = SigmaDefault;
		}


		private void trackThreshold_ValueChanged_1(object sender, EventArgs e)
		{
			lblThreshold.Text = trackThreshold.Value.ToString();
			Threshold = trackThreshold.Value;
		}

		private void trackSigma_ValueChanged(object sender, EventArgs e)
		{
			lblWindow.Text = trackSigma.Value.ToString();
			Sigma = trackSigma.Value;
		}
	}
}
